src/main/java/de/dennisguse/opentracks/services/announcement/VoiceAnnouncementUtils.java

```diff
@@ -61,7 +61,9 @@ static Spannable getAnnouncement(Context context, TrackStatistics trackStatistic
                 throw new RuntimeException("Not implemented");
         }
 
-        double distanceInUnit = distance.toKM_Miles(unitSystem);
+        // Total distance is announced to two decimal places, so round early to
+        // avoid rounding issues (see #1285)
+        double distanceInUnit = Math.round(distance.toKM_Miles(unitSystem) * 100) / 100.0;
```

Comment:
extract into a method?

Reply:
I was thinking about adjusting `appendDecimalUnit` to instead take a double, and a precision, which seemed to make sense.